### ISSUE TITLE
Weechat freezing .. every couple of hours - Now FIXED

### ISSUE THREAD

**Issue by laymonk** [weechat/weechat#1061]
I have been a long time user of weechat (on debian mostly) and within yakuake drop-down terminal.  

Recently, and to get rid of needless KDE libs and apps (I run xfce primarily) , I moved from yakuake to tilda terminal, and ever since then weechat freezes up on me regularly.

My regular terminal app is terminator, and weechat also freezes inside it.  So, that confirms to me that this is more likely a weechat problem than a problem with tilda or terminator.

I was originally running the version of weechat in Debian jessie (v1.0.1-1) .. Moving to v1.6-1 (in jessie-backports) has not resolved the problem.  My next consideration is to add your ppa and move up to v1.9-1, but before I do that I need to be sure I am not wasting my time needlessly.  

The crashes are frequent enough to make weechat unusable for me right now .. As a last resort, I am on the verge of  going back to irssi (which is not as good as weechat, IMHO) 

**How can I debug what is going on?**

There is a `[debug]` section in weechat.conf, but I cannot see any information on what needs to go in there (not in the manpages, nor on the web) .. 

One thing that changed recently was that `I got rid of avahi (libs and daemon) and kdelibs` .. does weechat depend on any these ?

FWIW, an `strace -f -p <pid>`  suggests this is possibly a socket problem .. I get the following repeatedly printed

```
[pid  8014] connect(17, {sa_family=AF_INET, sin_port=htons(0), sin_addr=inet_addr("0.0.0.0")}, 16) = -1 ECONNREFUSED (Connection refused)
[pid  8014] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid  8014] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid  8014] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid  8014] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid  8014] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid  8014] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid  8014] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid  8014] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid  8014] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid  8014] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid  8014] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid  8014] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid  8014] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid  8014] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid  8014] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid  8014] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid  8014] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid  8014] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid  8014] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid  8014] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid  8014] socket(PF_INET, SOCK_STREAM, IPPROTO_IP) = 16
[pid  8014] close(17)                   = 0
[pid  8014] fcntl(16, F_GETFL)          = 0x2 (flags O_RDWR)
[pid  8014] fcntl(16, F_SETFL, O_RDWR)  = 0
```
**I run with the following scripts and plugins**
```
perl scripts loaded: beep, highmon, iset
python scripts loaded: highlightxmpp, jabber
Plugins loaded: alias, aspell, buflist, charset, exec, fifo, guile, irc, javascript, \
   logger, lua, perl, python, relay, ruby, script, tcl, trigger, xfer
```

**Comment by laymonk** [comment]
I have now switched to v1.9-1 and the crashes continue .. and this time every few minutes .. 

I am strongly suspecting the now removed avahi as the issue, given the strace output .. But there is nothing in the weechat guides to suggest it's a dependency ..

**Comment by weechatter** [comment]
in most cases its a problem with scripts.
beep script is obsolete (use /trigger), its also recommend to use bitlbee instead of jabber.py.

remove/replace scripts and try again.

**Comment by laymonk** [comment]
@weechatter, thanks for responding .. let me try what you suggest .. 

But I do wonder why I never saw these things till now .. ie, why did weechat work happily for years running inside yakuake ?  

Also, further observation is that the freezes happen more often when the terminal in which weechat is running is a tab that loses focus due to a tab switch ..

**Comment by weechatter** [comment]
no, i mean: /trigger list

I installed tilda.  weechat does not freeze (running in a tab without focus)
runtime: 02:13:56

**Comment by laymonk** [comment]
@flashcode, thanks for all your assistance .. just been too busy to try and figure out and fix this .. 

here is what I have done so far:
* Removed the obsolete plugins you flagged above
* Now stuck on 1.9-1 (from your ppa) because if I downgrade the older 1.6-1 is unable to read the new config file format upgraded to by 1.9-1.  Which is OK, I run with 1.9-1 .
* I did a full desktop OS update this morning and rebooted (just to be sure I was done with any avahi lingering processes .. and such)

Still I get the occassional freeze-up (not as frequent as on last friday) .. but still causes me to lose messages and makes me even unaware that I crashed out until someone else notices, or I try to type or move between buffers ..

When weechat is frozen, running strace on the pid yields the following .. notice that, unlike the strace I pasted on friday, this one clearly says 'connection refused'  .. but I cannot tell what scripts, plugin, core process may be trying to setup this socket connection, and exactly what it is trying to do .. 

Was hoping you could suggest ways I can enable meaningful debugs to help .. I just cannot find the time to micro-troubleshoot this by manually adding scripts and plugins .. It's even doubtful that could be effective because the problem occurs intermittently and I have not figured out what event (or combination of events) triggers it

```
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] socket(PF_INET, SOCK_STREAM, IPPROTO_IP) = 18
[pid 26877] close(17)                   = 0
[pid 26877] fcntl(18, F_GETFL)          = 0x2 (flags O_RDWR)
[pid 26877] fcntl(18, F_SETFL, O_RDWR)  = 0
[pid 26877] connect(18, {sa_family=AF_INET, sin_port=htons(0), sin_addr=inet_addr("0.0.0.0")}, 16) = -1 ECONNREFUSED (Connection refused)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] select(0, NULL, NULL, NULL, {0, 100000}) = 0 (Timeout)
[pid 26877] socket(PF_INET, SOCK_STREAM, IPPROTO_IP) = 17
[pid 26877] close(18)                   = 0
[pid 26877] fcntl(17, F_GETFL)          = 0x2 (flags O_RDWR)
[pid 26877] fcntl(17, F_SETFL, O_RDWR)  = 0
[pid 26877] connect(17, {sa_family=AF_INET, sin_port=htons(0), sin_addr=inet_addr("0.0.0.0")}, 16) = -1 ECONNREFUSED (Connection refused)
```

**Comment by laymonk** [comment]
* I really have not found out why this started happening in the first place .. but the most likely culprit was a plugin that was trying to talk across a socket that does not exist .. I am suspecting `highlightxmpp`
* After removing the plugins @weechatter explained were obsolete, I believe I then intgroduced another issue which was that I installed bitlbee but never fully configured it, while still leaving `highlightxmpp` as a plugin.  An the connect requests in my last screen dump might have come from that.

**Bottom line**
* This is now fixed, and weechat (1.9) is not freezing in either terminator or tilda 

Thanks to @weechatter for his prompt responses and suggestions ..

**Comment by weechatter** [comment]
Weechat is single threaded. bad written scripts (a plugin is something completely different in weechat!) could easily freeze weechat. I will close the issue....
